packages/wallet-sdk/src/CoinbaseWalletProvider.ts

```diff
@@ -53,6 +54,16 @@ export class CoinbaseWalletProvider extends ProviderEventEmitter implements Prov
             storeSignerType(signerType);
             break;
           }
+          case 'wallet_sendCalls':
+          case 'wallet_sign': {
+            const ephemeralSigner = this.initSigner('scw');
+            await ephemeralSigner.handshake({ method: 'coinbase_handshake' }); // exchange session keys
+            const result = await ephemeralSigner.request(args); // send diffie-hellman encrypted request
+            await ephemeralSigner.cleanup(); // clean up (rotate) the ephemeral session keys
+            return result as T;
+          }
```

Comment:
I wonder if we should move this into a separate function/method to keep the request function small and make it easier to test?

Reply:
Also we should add tests for this method

Reply:
sg added tests on ephemeral signer behaviour!

hmmm i'm actually kinda leaning towards keeping the ephemeral signer logic in place, cause this is the only place that we will ever call this logic, so i'm feeling a bit hesitant on abstracting it to a separate func, wdyt

Reply:
Sounds good for now.